### DISCUSSION TITLE
Referencing in JavaScript - Usage and Abusage

### DISCUSSION THREAD

**Discussion by Martleby** [tailwindlabs/tailwindcss#4916] [Help]
Hi All,

I'm using some third party Vue components (ie Vue ChartJS) that accept colours as hex or rgb. I thought it would be neat to supply Tailwind's color classes instead so I followed along with the docs https://tailwindcss.com/docs/configuration#referencing-in-java-script

This is all working fine... but since I don't really understand what this does or how it works, I would be grateful if somebody could answer these questions:

- I use JIT mode to keep my CSS nice and small. Does enabling the above run the risk of bloating out my JavaScript instead?
- The docs mention the use of babel-plugin-preval. I've read the docs and watched the video twice and I'm still unclear what this does, why it's needed, and how to configure and use it (in the context of a Laravel app using Mix).
- Is there a better way of doing what I what to do?

Cheers,
Martin.

**Comment by deadcoder0904** [comment]
> * I use JIT mode to keep my CSS nice and small. Does enabling the above run the risk of bloating out my JavaScript instead?

I don't think so but not sure. There's also a way to whitelist some classes which you are using to not purge them so you will be whitelisting the classes you reference in your Vue ChartJS so as to not purge it.

> * The docs mention the use of babel-plugin-preval. I've read the docs and watched the video twice and I'm still unclear what this does, why it's needed, and how to configure and use it (in the context of a Laravel app using Mix).

This [README](https://github.com/kentcdodds/babel-plugin-preval/blob/main/other/EXAMPLES.md) explains it better. The plugin basically pre-evaluate stuff at build time rather than runtime. Again, the README explains it better:

> You need to do some dynamic stuff, but don't want to do it at runtime. Or maybe you want to do stuff like read the filesystem to get a list of files and you can't do that in the browser.

The 2nd line explains it much better I think. You can't perform `fs` operations in the browser since it's a Node.js module so you can pre-evaluate that code & it gets outputted into static code.

> * Is there a better way of doing what I what to do?

Nope, I don't think so.

I'm a Tailwind noob so take my answers with a grain of salt :)
